akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.scaladsl
+
+import scala.concurrent.Future
+import org.scalatest.{ WordSpec, Matchers }
+import akka.actor.ActorSystem
+
+class HttpServerExampleSpec extends WordSpec with Matchers {
+
+  "binding-example" in {
+    import akka.stream.ActorFlowMaterializer
+    import akka.stream.scaladsl._
+    import akka.http.scaladsl.Http
+
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorFlowMaterializer()
+
+    val serverSource: Source[Http.IncomingConnection, Future[Http.ServerBinding]] =
+      Http().bind(interface = "localhost", port = 8080)
+    val bindingFuture: Future[Http.ServerBinding] =
+      serverSource.to(Sink.foreach { connection => // foreach materializes the source
```

Comment:
Added a ticket for a "parallel" version of foreach that might be more suited for this: https://github.com/akka/akka/issues/17461

(we can discuss there)